File: subprojects/frida-python/examples/open_service/dtx/opengl.py
import sys

import frida


def on_message(message):
    print("on_message:", message)


device = frida.get_usb_device()

opengl = device.open_service("dtx:com.apple.instruments.server.services.graphics.opengl")
opengl.on("message", on_message)
opengl.request(
    {
        "method": "setSamplingRate:",
        "args": [5.0],
    }
)
opengl.request(
    {
        "method": "startSamplingAtTimeInterval:",
        "args": [0.0],
    }
)

sys.stdin.read()

File: subprojects/frida-node/examples/open_service/dtx/opengl.ts
import frida from "frida";

let opengl: frida.Service | null = null;

async function main() {
    const device = await frida.getUsbDevice();

    opengl = await device.openService("dtx:com.apple.instruments.server.services.graphics.opengl");
    opengl.message.connect(onMessage);
    await opengl.request({ method: "setSamplingRate:", args: [5.0] });
    await opengl.request({ method: "startSamplingAtTimeInterval:", args: [0.0] });
}

function onMessage(message: any) {
    console.log("onMessage:", message);
}

main()
    .catch(e => {
        console.error(e);
    });
